internal/catalogmetadata/filter/bundle_predicates.go

```diff
@@ -77,9 +78,9 @@ func LegacySuccessor(installedBundle *catalogmetadata.Bundle) Predicate[catalogm
 			}
 		}
 		if candidateBundleEntry.SkipRange != "" {
-			installedBundleVersion, _ := installedBundle.Version()
-			skipRange, _ := bsemver.ParseRange(candidateBundleEntry.SkipRange)
-			if installedBundleVersion != nil && skipRange != nil && skipRange(*installedBundleVersion) {
+			installedBundleVersion, vErr := bsemver.Parse(installedBundle.Version)
+			skipRange, srErr := bsemver.ParseRange(candidateBundleEntry.SkipRange)
+			if vErr == nil && srErr == nil && skipRange(installedBundleVersion) {
```

Comment:
Both the before and after code just silently ignore edges if the installed bundle version or the skipRange cannot be parsed.

I'm tempted to actually fail here if the bundle version or skip range fail to parse.

Thoughts?

Reply:
What is the impact to non-semver versions if we fail here? Do we know how many packages use versioning schemes incompatible with semver in the commonly used catalogs?

Reply:
Pretty sure FBC requires valid semver and `opm validate` checks both places. So in theory, this should never fail, and if it does, it is because someone has invalid FBC.

Which is why I think I'd rather just fail here. If we ignore it, it's this sorta weird fence-straddling position of accepting it, but then not doing anything with it.

Reply:
If there is validation that the versions must be valid semver then I am fine with failing here

Reply:
Looks like it would be a significant change to bubble errors up through the predicate API. Probably better to handle this in a follow-up. I think we should do one of the following:
1. Update the predicate API to return `(bool, error)`. This is a big change because _lots_ of stuff implements the predicate interface.
2. Update the bundle-related predicates to use a types where the bundle version and channel skipRanges are pre-parsed. That way we eliminate the calls inside the predicate that could return an error.